alchemy/src/neon/project.ts

```diff
@@ -298,3 +331,125 @@ export const NeonProject = Resource(
     }
   },
 );
+
+/**
+ * References an existing Neon project by name. You can use this to reference an existing NeonProject without adopting it.
+ *
+ * @example
+ * const project = await NeonProjectRef({
+ *   name: "my-project"
+ * });
+ *
+ * @warning Project names must be unique when referencing existing projects. If multiple projects are found with the same name, an error will be thrown.
+ */
+export async function NeonProjectRef(
+  props: NeonProjectProps & { name: string },
+): Promise<NeonProject> {
+  return await fetchProject(createNeonApi(props), props);
+}
+
+async function fetchProject(
+  api: NeonClient,
+  props: NeonProjectProps & { name: string },
+): Promise<NeonProject> {
+  const project = await api
+    .listProjects({
+      query: { search: props.name },
+    })
+    .then(({ data: { projects } }) => {
+      if (!projects.length) {
+        throw new NeonProjectNotFound(props.name);
+      }
+      if (projects.length > 1) {
+        throw new Error(
+          `Multiple projects found with name "${props.name}". Name must be unique when referencing existing projects.`,
+        );
+      }
+      return projects[0];
+    });
+  const endpointsPromise = api
```

Comment:
Declared this and `rolesPromise` separately from awaiting them in `Promise.all` at the end as a slight optimization since they're not part of the branch > databases > connection URI waterfall.